memcached/src/main/java/com/yahoo/ycsb/db/MemcachedCompatibleClient.java

```diff
@@ -0,0 +1,239 @@
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+import java.util.Vector;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+/**
+ * Memcached-compatible base class.
+ */
+public abstract class MemcachedCompatibleClient extends DB {
+
+  private final Logger logger = Logger.getLogger(getClass());
+
+  protected static final ObjectMapper MAPPER = new ObjectMapper();
+
+  /**
+   * The MemcachedClient implementation that will be used to communicate
+   * with the memcached server.
+   */
+  private MemcachedClient client;
+
+  private boolean checkOperationStatus;
+  private long shutdownTimeoutMillis;
+  private int objectExpirationTime;
+
+  private static final String TEMPORARY_FAILURE_MSG = "Temporary failure";
+  private static final String CANCELLED_MSG = "cancelled";
+
+  public static final String SHUTDOWN_TIMEOUT_MILLIS_PROPERTY =
+      "memcached.shutdownTimeoutMillis";
+  public static final String DEFAULT_SHUTDOWN_TIMEOUT_MILLIS = "30000";
+
+  public static final String OBJECT_EXPIRATION_TIME_PROPERTY =
+      "memcached.objectExpirationTime";
+  public static final String DEFAULT_OBJECT_EXPIRATION_TIME =
+      String.valueOf(Integer.MAX_VALUE);
+
+  public static final String CHECK_OPERATION_STATUS_PROPERTY =
+      "memcached.checkOperationStatus";
+  public static final String CHECK_OPERATION_STATUS_DEFAULT = "true";
+
+  /**
+   * @returns Underlying Memcached protocol client, implemented by
+   *     SpyMemcached.
+   */
+  protected MemcachedClient memcachedClient() {
+    return client;
+  }
+
+  @Override
+  public void init() throws DBException {
+    try {
+      client = createMemcachedClient();
+      checkOperationStatus = Boolean.parseBoolean(
+          getProperties().getProperty(CHECK_OPERATION_STATUS_PROPERTY,
+                                      CHECK_OPERATION_STATUS_DEFAULT));
+      objectExpirationTime = Integer.parseInt(
+          getProperties().getProperty(OBJECT_EXPIRATION_TIME_PROPERTY,
+                                      DEFAULT_OBJECT_EXPIRATION_TIME));
+      shutdownTimeoutMillis = Integer.parseInt(
+          getProperties().getProperty(SHUTDOWN_TIMEOUT_MILLIS_PROPERTY,
+                                      DEFAULT_SHUTDOWN_TIMEOUT_MILLIS));
+    } catch (Exception e) {
+      throw new DBException(e);
+    }
+  }
+
+  protected abstract MemcachedClient createMemcachedClient() throws Exception;
+
+  @Override
+  public Status read(
+      String table, String key, Set<String> fields,
+      HashMap<String, ByteIterator> result) {
+    key = createQualifiedKey(table, key);
+    try {
+      GetFuture<Object> future = memcachedClient().asyncGet(key);
+      Object document = future.get();
+      if (document != null) {
+        fromJson((String) document, fields, result);
+      }
+      return Status.OK;
+    } catch (Exception e) {
+      logger.error("Error encountered for key: " + key, e);
+      return new Status("ERROR_READ", e.toString());
```

Comment:
Status codes are already going to be grouped by operation time ie. READ, UPDATE, etc. You are also logging the error so the user will see the exception in the output. So, it is not necessary to record a new status of "ERROR_READ", ```Status.ERROR``` should suffice.